com.unity.shadergraph/Editor/Data/Graphs/ColorShaderProperty.cs

```diff
@@ -41,25 +49,34 @@ public ColorMode colorMode
             get => m_ColorMode;
             set => m_ColorMode = value;
         }
-        
+
         internal override AbstractMaterialNode ToConcreteNode()
         {
             return new ColorNode { color = new ColorNode.Color(value, colorMode) };
         }
 
         internal override PreviewProperty GetPreviewMaterialProperty()
         {
-            return new PreviewProperty(propertyType)
+            UnityEngine.Color propColor = value;
+            if (colorMode == ColorMode.HDR)
+            {
+                if (PlayerSettings.colorSpace == ColorSpace.Linear)
+                    propColor = propColor.gamma;
+            }
+            // we use Vector4 type to avoid all of the automatic color conversions of PropertyType.Color
+            return new PreviewProperty(PropertyType.Color)
```

Comment:
I would prefer using PropertyType.Vector4 and do all the colorspace conversions in this function; otherwise we are converting to gamma space above just to counteract the automatic gamma->linear conversion that happens via PropertyType.Color.  Fewer conversions to mess with your values, and code that makes more sense.

Reply:
`

            UnityEngine.Color propColor = value;
            if (colorMode == ColorMode.Default)
            {
                if (PlayerSettings.colorSpace == ColorSpace.Linear)
                    propColor = propColor.linear;
            }
            else if (colorMode == ColorMode.HDR)
            {
                // conversion from linear to active color space is handled in the shader code (see PropertyNode.cs)
            }

            // we use Vector4 type to avoid all of the automatic color conversions of PropertyType.Color
            return new PreviewProperty(PropertyType.Vector4)
            {
                name = referenceName,
                vector4Value = propColor
            };

`